numpy/lib/npyio.py

```diff
@@ -112,7 +112,7 @@ def zipfile_factory(file, *args, **kwargs):
     pathlib.Path objects. `args` and `kwargs` are passed to the zipfile.ZipFile
     constructor.
     """
-    if not hasattr(file, 'read'):
```

Comment:
This one looks suspect - are you sure this isn't used for both reading and writing? I'd be inclined to check for either member here.

Reply:
I think you're right. I will remove that one.